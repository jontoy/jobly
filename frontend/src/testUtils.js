import React from "react";
import UserContext from "./userContext";

const demoUser = {
  username: "test",
  first_name: "test",
  last_name: "mctest",
  email: "test@test.net",
  photo_url: null,
  jobs:[]
};

const UserProvider = ({ children, currentUser = demoUser, appliedIds = [] }) => (
  <UserContext.Provider value={{currentUser, appliedIds}}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
