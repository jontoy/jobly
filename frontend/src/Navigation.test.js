import React from 'react';
import { render } from '@testing-library/react';
import Navigation from './Navigation';
import { MemoryRouter } from 'react-router-dom';
import {UserProvider} from './testUtils';

it('renders without crashing', () => {
    render(
    <MemoryRouter>
        <UserProvider>
            <Navigation />
        </UserProvider>
    </MemoryRouter>
    );
});
  
it('matches snapshot when user is logged in', () => {
      const {asFragment} = render(
        <MemoryRouter>
            <UserProvider>
                <Navigation />
            </UserProvider>
        </MemoryRouter>
        );
    expect(asFragment()).toMatchSnapshot();
});

it('matches snapshot when user is logged out', () => {
    const {asFragment} = render(
      <MemoryRouter>
          <UserProvider currentUser={null}>
              <Navigation />
          </UserProvider>
      </MemoryRouter>
      );
  expect(asFragment()).toMatchSnapshot();
});