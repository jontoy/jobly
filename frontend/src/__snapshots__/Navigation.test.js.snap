// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when user is logged in 1`] = `
<DocumentFragment>
  <nav
    class="Navigation navbar-expand navbar navbar-dark bg-primary"
  >
    <a
      aria-current="page"
      class="Navigation-link navbar-brand active"
      href="/"
    >
      Jobly
    </a>
    <ul
      class="navbar-nav ml-auto"
    >
      <li
        class="nav-item"
      >
        <a
          class="Navigation-link"
          href="/login"
        >
          Login
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`matches snapshot when user is logged out 1`] = `
<DocumentFragment>
  <nav
    class="Navigation navbar-expand navbar navbar-dark bg-primary"
  >
    <a
      aria-current="page"
      class="Navigation-link navbar-brand active"
      href="/"
    >
      Jobly
    </a>
    <ul
      class="navbar-nav ml-auto"
    >
      <li
        class="nav-item"
      >
        <a
          class="Navigation-link"
          href="/login"
        >
          Login
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
